src/NUnitFramework/framework/Internal/Results/TestSuiteResult.cs

```diff
@@ -112,46 +194,66 @@ public override IList<ITestResult> Children
         /// <param name="result">The result to be added</param>
         public virtual void AddResult(ITestResult result)
         {
-            Children.Add(result);
+#if PARALLEL
+            var childrenAsConcurrentQueue = Children as ConcurrentQueue<ITestResult>;
+            if (childrenAsConcurrentQueue != null)
+                childrenAsConcurrentQueue.Enqueue(result);
+            else
+#endif
+            {
+                var childrenAsIList = Children as IList<ITestResult>;
+                if (childrenAsIList != null)
+                    childrenAsIList.Add(result);
+                else
+                    throw new NotSupportedException("cannot add results to Children");
```

Comment:
But we're in TestSuiteResult here and _children can't be anything. Its actual type is known to us. If PARALLEL is defined, the first cast will always succeed. If not, the second cast will always succeed.

The field _children doesn't even exist in the base or in TestCaseResult but is private here and the AddResult method only exists here. What you say would be true if we were talking about the Children of the result that is being added, but here we are dealing with `this` result and we know the implementation type.